docs/docs/dev_docs/contracts/portals/main.md

```diff
@@ -0,0 +1,331 @@
+
+:::info
+Error handling for cross chain messages is handled by the application contract and not the protocol. The protocol only delivers the messages, it does not ensure that they are executed successfully.
+:::
+
+### Cancellations
+
+A special type of errors is underpriced transactions, here meaning that message is inserted on L1, but the fee available to the sequencer is too low to be included in a rollup block.
+
+For the case of token bridges, this could lead to funds being locked in the bridge forever, as the funds are locked but the message never arrives on L2 to mint the tokens. To address this, the `Inbox` supports cancelling messages after a deadline. However, this must be called by the portal itself, as it will need to "undo" the state changes is made (for example by sending the tokens back to the user).
+
+As this requires logic on the portal itself, it is not something that the protocol can enforce. It must be supported by the application builder when building the portal.
+
+The portal can call the `cancelL2Message` at the `Inbox` when `block.timestamp > deadline` for the message.
+
+```solidity title="IInbox.sol"
+function cancelL2Message(
+    DataStructures.L1ToL2Msg memory _message, 
+    address _feeCollector
+) external returns (bytes32 entryKey);
+```
+
+Building on our token example from earlier, this can be called like:
+
+```solidity title="TokenPortal.sol"
+function cancelL1ToAztecMessage(
+    bytes32 _to,
+    uint256 _amount,
+    uint32 _deadline,
+    bytes32 _secretHash,
+    uint64 _fee
+  ) external returns (bytes32) {
+    IInbox inbox = registry.getInbox();
+    DataStructures.L1Actor memory l1Actor = DataStructures.L1Actor(address(this), block.chainid);
+    DataStructures.L2Actor memory l2Actor = DataStructures.L2Actor(l2TokenAddress, 1);
+    DataStructures.L1ToL2Msg memory message = DataStructures.L1ToL2Msg({
+      sender: l1Actor,
+      recipient: l2Actor,
+      content: Hash.sha256ToField(
+        abi.encodeWithSignature("mint(uint256,bytes32,address)", _amount, _to, msg.sender)
+        ),
+      secretHash: _secretHash,
+      deadline: _deadline,
+      fee: _fee
+    });
+    bytes32 entryKey = inbox.cancelL2Message(message, address(this));
+    // Ensures that `msg.sender == canceller` by using `msg.sender` in the hash computation.
+    underlying.transfer(msg.sender, _amount);
+    return entryKey;
+  }
+```
+
+The example above ensure that the user can cancel his message if it is underpriced.
+
+### Designated caller
+A designated caller is the ability to specify who should be able to call a function that consumes a message. This is useful for ordering of batched messages, as we will see in a second.  
+
+When doing multiple cross-chain calls as one action it is important to consider the order of the calls. Say for example, that you want to do a uniswap trade on L1 because you are a whale and slippage on L2 is too damn high. 
```

Comment:
```suggestion
When performing multiple cross-chain calls in one action it is important to consider the order of the calls. Say for example, that you want to perform a uniswap trade on L1 because you are a whale and slippage on L2 is too damn high. 
```